CONTRIBUTING.md

```diff
@@ -29,30 +31,140 @@ and maintained by Google](https://opensource.google.com/docs/cla/#why), not to
      may need to be broken into multiple smaller tests.
 
-1. The pull request should include both the suggested feature profile textproto 
+1. The pull request should include both the suggested feature profile textproto
  additions, as well as any relevant additions to tests. Tests should be written
  in golang using the [ONDATRA](https://github.com/openconfig/ondatra) framework.
 
 1. The automated CI running against each pull request will check the pull
  request for compliance.  The author should resolve any issues found by CI.
 
-1. One or more peers in the community may review the pull request.   
+1. One or more peers in the community may review the pull request.
 
 1. A feature profile repository maintainer will be reponsible for a final review
-and approval.  Only a feature repository maintainer can merge a pull request to 
+and approval.  Only a feature repository maintainer can merge a pull request to
 the main branch.
   
-The aim of this process is maintain the model quality and approach that OpenConfig 
+The aim of this process is maintain the model quality and approach that OpenConfig
 working group has strived for since its inception in 2014. Thank you for your contributions!
+
+## Test development style guide
+
+### Directory Organization
+
+The directory tree is organized as follows:
+
+* `cloudbuild/` contains google cloud build scripts for running virtual
+    routers in containers on [KNE](https://github.com/google/kne)
+* `feature/` contains definition and tests of feature profiles.
+* `internal/` contains packages used by feature profile tests.
+* `proto/`  contains protobuf files for feature profiles.
+* `tools/` contains code used for CI checks.
+* `topologies/` contains the testbed topology definitions.
+* `yang/` contains autogenerated go structures.
+
+### Test Suite Organization
+
+Test suites should be placed in subdirectories formatted like `feature/<featurename>/[<sub-feature>/]tests/<testname>/main_test.go`, for example:
+
+* `feature/interfaces/` contains feature profiles for interfaces.
+* `feature/interfaces/README.md` - documents the interface feature profile.
+* `feature/interfaces/tests/` contains the interfaces test suite.
+* `feature/bgp/` container for BGP feature profiles and test suites.
+* `feature/bgp/addpath` contains feature profiles related to BGP.
+* `feature/bgp/addpath/README.md` documents the BGP feature profile.
+* `feature/bgp/addpath/tests` contains test code related to BGP Add Path features.
+* `internal/deviations` contains code which overrides standard tests where
+    there are known issues in a DUT.
+
+Within each test directory, a README.md should document the test suite.  The `*.go` files should be named after an appropriate
+[github featureprofiles project](https://github.com/orgs/openconfig/projects/2/views/1?filterQuery=)
+item. For example:
+
+* `feature/interfaces/tests/rt_5_1_singleton_test/README.md` - documentation for the test suite
```

Comment:
There also seems to be a consensus that I heard that the XX-1.2 designations are going to be removed, so I removed them. e.g. `feature/interfaces/ate_tests/singleton_test`.

Reply:
Ok, will leave XX-#.# the identifiers  for the moment.